lib/xml_wrap.ml

```diff
@@ -31,15 +31,49 @@ module type T = sig
   val map : ('a -> 'b) -> 'a tlist -> 'b tlist
 end
 
+(* [T'] extends [T] with a type constructor [ft] that represents
+   "pseudo-functions" to be applied on wrapped elements. [(a, b) ft]
+   is the type of pseudo-functions that transform inhabitants of [a t]
+   into inhabitants of [b t]. Such pseudo-functions are applied via
+   [fmap]. For example, ('a, 'b) ft can be instantiated with ('a ->
+   'b) Eliom_lib.shared_value, thus enabling operations on shared
+   values. Such operations would be impossible to perform via plain
+   functions. *)
+
+module type T' = sig
+  type 'a t
+  val return : 'a -> 'a t
+
+  type ('a, 'b) ft
+  val fmap :
+    ('a, 'b) ft -> 'a t -> 'b t
+
+  type 'a tlist
+  val nil : unit -> 'a tlist
+  val singleton : 'a t -> 'a tlist
+  val cons : 'a t -> 'a tlist -> 'a tlist
+  val append : 'a tlist -> 'a tlist -> 'a tlist
+  val map : ('a, 'b) ft -> 'a tlist -> 'b tlist
+end
+
```

Comment:
No need to have two versions, almost nobody except eliom/jsoo is using T directly. Given the "most complete" version, you can trivially define the other with `T with type ('a, 'b) ft = 'a -> 'b` so it's not that much of an issue.